File: app/src/main/java/com/mrgames13/jimdo/feinstaubapp/CommonObjects/Sensor.java
package com.mrgames13.jimdo.feinstaubapp.CommonObjects;

public class Sensor {

    //Konstanten

    //Variablen als Objekte

    //Variablen
    private String id = "no_id";
    private String name = "unknown";
    private int color = 0;

    public Sensor() {}

    public Sensor(String id, String name, int color) {
        this.id = id;
        this.name = name;
        this.color = color;
    }

    public void setId(String id) {
        this.id = id;
    }
    public String getId() {
        return id;
    }

    public void setName(String name) {
        this.name = name;
    }
    public String getName() {
        return name;
    }

    public void setColor(int color) {
        this.color = color;
    }
    public int getColor() {
        return color;
    }
}

File: app/src/main/java/com/mrgames13/jimdo/feinstaubapp/CommonObjects/Sensor.kt
/*
 * Copyright © 2019 Marc Auberer. All rights reserved.
 */

package com.mrgames13.jimdo.feinstaubapp.CommonObjects

import java.io.Serializable

class Sensor: Comparable<Any>, Serializable {

    // Variables
    var chipID = "no_id"
    var name = "unknown"
    var color = 0

    constructor()

    constructor(id: String, name: String, color: Int) {
        this.chipID = id
        this.name = name
        this.color = color
    }

    fun setId(id: String) {
        this.chipID = id
    }

    override operator fun compareTo(other: Any): Int {
        val other_sensor = other as Sensor
        return name.compareTo(other_sensor.name)
    }
}